File: backend/java/src/main/java/com/example/bookapi/CreateBookDto.java
package com.example.bookapi;

import jakarta.validation.constraints.NotBlank;

public class CreateBookDto {
    @NotBlank
    private String title;
    @NotBlank
    private String author;
    @NotBlank
    private String description;
    @NotBlank
    private String coverImageUrl;

    public CreateBookDto() {}

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCoverImageUrl() {
        return coverImageUrl;
    }

    public void setCoverImageUrl(String coverImageUrl) {
        this.coverImageUrl = coverImageUrl;
    }
}

File: backend/csharp/Models/CreateBookDto.cs
using System.ComponentModel.DataAnnotations;

namespace BookApi;

public class CreateBookDto
{
    [Required]
    public required string Title { get; set; }
    [Required]
    public required string Author { get; set; }
    [Required]
    public required string Description { get; set; }
    [Required]
    public required string CoverImageUrl { get; set; }
}